bundles/org.openhab.binding.philipsair/src/main/java/org/openhab/binding/philipsair/internal/discovery/PhilipsAirUpnpDiscoveryParticipant.java

```diff
@@ -0,0 +1,136 @@
+    private boolean isAutoDiscoveryEnabled = true;
+
+    @Activate
+    protected void activate(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    @Modified
+    protected void modified(ComponentContext componentContext) {
+        activateOrModifyService(componentContext);
+    }
+
+    private void activateOrModifyService(ComponentContext componentContext) {
+        Dictionary<String, @Nullable Object> properties = componentContext.getProperties();
+        String autoDiscoveryPropertyValue = (String) properties.get("enableAutoDiscovery");
+        if (autoDiscoveryPropertyValue != null && autoDiscoveryPropertyValue.length() != 0) {
+            isAutoDiscoveryEnabled = Boolean.valueOf(autoDiscoveryPropertyValue);
+        }
+    }
+
+    @Override
+    public Set<ThingTypeUID> getSupportedThingTypeUIDs() {
+        return SUPPORTED_THING_TYPES_UIDS;
+    }
+
+    @Override
+    public @Nullable DiscoveryResult createResult(RemoteDevice device) {
+        if (!isAutoDiscoveryEnabled) {
+            return null;
+        }
+
+        ThingUID uid = getThingUID(device);
+        if (uid != null) {
+            logger.trace("Creating with uid {}", uid.getAsString());
+            Map<String, Object> properties = new HashMap<>();
+            properties.put(PhilipsAirConfiguration.CONFIG_HOST, device.getIdentity().getDescriptorURL().getHost());
+            properties.put(PhilipsAirConfiguration.CONFIG_DEF_DEVICE_UUID,
+                    device.getIdentity().getUdn().getIdentifierString());
+
+            DiscoveryResult result = DiscoveryResultBuilder.create(uid).withProperties(properties)
+                    .withLabel("Philips " + device.getDetails().getModelDetails().getModelName() + " "
+                            + device.getDetails().getModelDetails().getModelNumber())
+                    .withRepresentationProperty(PhilipsAirConfiguration.CONFIG_DEF_DEVICE_UUID).build();
+
+            logger.info("DiscoveryResult with uid {} label : {} ", result.getThingUID().getAsString(),
+                    result.getLabel());
+            return result;
+        } else {
+            return null;
+        }
+    }
+
+    @Override
+    public @Nullable ThingUID getThingUID(RemoteDevice device) {
+        DeviceDetails details = device.getDetails();
+        ModelDetails modelDetails = null;
+        String modelName = null;
+
+        if (details == null || (modelDetails = details.getModelDetails()) == null
+                || !PhilipsAirBindingConstants.DISCOVERY_UPNP_MODEL
+                        .equalsIgnoreCase(modelName = modelDetails.getModelName())) {
+            logger.trace("Device not recognized {}", device.toString());
+            return null;
+        }
+
+        ThingTypeUID thingType = THING_TYPE_UNIVERSAL;
+        if (PhilipsAirBindingConstants.SUPPORTED_MODEL_NUMBER_AC2889_10
+                .startsWith(modelDetails.getModelNumber().toLowerCase())) {

```

Comment:
cache `modelDetails.getModelNumber().toLowerCase()` in a local variable so you can reuse it.

Reply:
Good point